src/include/simeng/OS/SyscallHandler.hh

```diff
@@ -103,13 +105,106 @@ struct linux_dirent64 {
   char* d_name;       // Filename (null-terminated)
 };
 
-/** A Linux kernel syscall emulation implementation, which mimics the
-   responses to Linux system calls. */
+/** The types of changes that can be made to values within the process state. */
+enum class ChangeType { REPLACEMENT, INCREMENT, DECREMENT };
+
+/** A structure describing a set of changes to the process state. */
+struct ProcessStateChange {
+  /** Type of changes to be made */
+  ChangeType type;
+  /** Registers to modify */
+  std::vector<Register> modifiedRegisters;
+  /** Values to set modified registers to */
+  std::vector<RegisterValue> modifiedRegisterValues;
+  /** Memory address/width pairs to modify */
+  std::vector<MemoryAccessTarget> memoryAddresses;
+  /** Values to write to memory */
+  std::vector<RegisterValue> memoryAddressValues;
+};
+
+/** The result from a handled syscall. */
+struct SyscallResult {
+  /** Whether execution should halt. */
+  bool fatal;
+  /** Id of the syscall to aid exception handler processing. */
+  uint64_t syscallId;
+  // The unique ID of the core associated with the syscall
+  uint64_t coreId;
+  /** Any changes to apply to the process state. */
+  ProcessStateChange stateChange;
+};
+
+/** A struct to hold information used as arguments to a syscall. */
+struct SyscallInfo {
+  // The ID of the syscall
+  uint64_t syscallId;
+  // The unique sequenceID of the instructions triggering the syscall
```

Comment:
What is a sequenceID, more information regarding what it is or represents would be really helpful

Reply:
The sequenceID is the instructions ID as described in `Instruction.hh`
```
/** This instruction's sequence ID; a higher ID represents a chronologically
   * newer instruction. */
  uint64_t sequenceId_;
```
Not sure how much more explicit "The unique sequenced of the instruction" could be here? Noted this has a typo tho so will change that now.

Reply:
My reasoning was that it isn't apparent on first glance what an instruction's sequence id is. But since the comment in Instruction.hh is also similar , i see that there is no point in changing. However would it better to rename the variable to `instrSeqId` just to clarify the intent and add a space between "sequenceID"

Reply:
Renamed variable and added space